src/nlsmpb_model.jl

```diff
@@ -0,0 +1,253 @@
+using MathProgBase
+
+include("mpb_model.jl")
+
+export MathProgNLSModel,
+       reset!,
+       obj, grad, grad!,
+       cons, cons!, jac_coord, jac, jprod, jprod!, jtprod, jtprod!,
+       hess_coord, hess, hprod, hprod!,
+       residual!, jac_residual, jprod_residual!, jtprod_residual!,
+       jac_op_residual, hess_residual, hprod_residual!, cons, cons!, jac_coord,
+       jac, jprod, jprod!, jtprod, jtprod!, hess, hess_coord, hprod, hprod!
+
+mutable struct MathProgNLSModel <: AbstractNLSModel
+  meta :: NLPModelMeta
+  nls_meta :: NLSMeta
```

Comment:
Are we moving towards merging both kinds of models into one then?

Reply:
Not yet, but this model will transition smoothly, it seems.

Reply:
Ok